### PR TITLE
[FIX] google_drive: adpating to new framework JS

### PR DESCRIPTION
**Pull request by kebeclibre**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/google_drive/static/src/js/gdrive.js [hunk 1]
```diff
@@ -5,56 +5,54 @@ var data = require('web.data');
 var Sidebar = require('web.Sidebar');
 
 Sidebar.include({
-    init: function () {
-        var self = this;
-        var ids;
+    // TO DO: clean me in master
```
Comment (aab-odoo) [inline addons/google_drive/static/src/js/gdrive.js]: what should be cleaned?

**Comment by kebeclibre** [comment]
@aab-odoo 
The test has been adapted in order for the new route to not be called

File: addons/google_drive/static/tests/gdrive_test.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+var FormView = require('web.FormView');
+var testUtils = require('web.test_utils');
+var GoogleDriveSideBar = require('google_drive.sidebar');
+
+var createView = testUtils.createView;
+
+// Keep a reference to the Sidebar's function
+// that checks google drive stuff via RPC
+// Then overriding that same function in the Sidebar to avoid
```
Comment (aab-odoo) [inline addons/google_drive/static/tests/gdrive_test.js]: override

File: addons/google_drive/static/tests/gdrive_test.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+// other tests to break because of the new route
+var _addGoogleDocItems_original = GoogleDriveSideBar.prototype._addGoogleDocItems;
+
```
Comment (aab-odoo) [inline addons/google_drive/static/tests/gdrive_test.js]:
```
var _addGoogleDocItemsMocked = function (model, resID) {
    return $.when();
};
GoogleDriveSidebar.prototype._addGoogleDocItems = _addGoogleDocItemsMocked;
```

File: addons/google_drive/static/tests/gdrive_test.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+var testUtils = require('web.test_utils');
+var GoogleDriveSideBar = require('google_drive.sidebar');
+
+var createView = testUtils.createView;
+
+// Keep a reference to the Sidebar's function
+// that checks google drive stuff via RPC
+// Then overriding that same function in the Sidebar to avoid
+// other tests to break because of the new route
+var _addGoogleDocItems_original = GoogleDriveSideBar.prototype._addGoogleDocItems;
```
Comment (aab-odoo) [inline addons/google_drive/static/tests/gdrive_test.js]: var _addGoogleDocItemsOriginal

File: addons/google_drive/static/tests/gdrive_test.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+GoogleDriveSideBar.include({_addGoogleDocItems: function (model, resID) {
+        return $.when();
+    }
+});
+
+QUnit.module('gdrive_integration', {
+    beforeEach: function () {
+        // For our test to work, the _addGoogleDocItems function needs to be the original
+        GoogleDriveSideBar.include({_addGoogleDocItems: _addGoogleDocItems_original});
```
Comment (aab-odoo) [inline addons/google_drive/static/tests/gdrive_test.js]: GoogleDriveSidebar.prototype._addGoogleDocItems = _addGoogleDocItemsOriginal;

File: addons/google_drive/static/tests/gdrive_test.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                    name: {string: 'Name', type:'char'}
+                },
+                records: [],
+            }
+        };
+    },
+
+    afterEach: function() {
+        GoogleDriveSideBar.include({_addGoogleDocItems: function (model, resID) {
+        return $.when();
```
Comment (aab-odoo) [inline addons/google_drive/static/tests/gdrive_test.js]: GoogleDriveSidebar.prototype._addGoogleDocItems = _addGoogleDocItemsMocked;

**Comment by kebeclibre** [comment]
closed at cdd3148d5ea83b49f3b6b25caae1f56d952e4459

**Pull request closed without merging**